CMakeLists.txt

```diff
@@ -124,13 +124,17 @@ if(ROBOTOLOGY_ENABLE_DYNAMICS)
   if(ROBOTOLOGY_USES_MATLAB OR ROBOTOLOGY_USES_OCTAVE)
     find_or_build_package(yarp-matlab-bindings)
   endif()
+  find_or_build_package(qpOASES)
   if(ROBOTOLOGY_USES_MATLAB)
-    find_or_build_package(qpOASES)
     if(NOT ROBOTOLOGY_NOT_USE_SIMULINK)
       find_or_build_package(WBToolbox)
       find_or_build_package(whole-body-controllers)
     endif()
-  endif()  
+  endif()
+  find_or_build_package(osqp)
+  find_or_build_package(OsqpEigen)
+  find_or_build_package(UnicyclePlanner)
+  find_or_build_package(icubWalkingDCM)
```

Comment:
I think we can just leave this line (or at limit the UnicyclePlanner). All other projects will be automatically downloaded and compiled as dependency listed.

Reply:
I see, even if I prefer to have the full list available in the ``CMakeLists`` to make sure at a first sight what is compiled.

Reply:
I see, but I would at least avoid the external (`osqp` and `qpOASES`).  Furthermore, for consistency I would put all the non-conditional `find_or_build_package`'s call before the first  `if(ROBOTOLOGY_USES_GAZEBO)`.